### PR TITLE
Core 4600 kafka auth attempt2

### PR DESCRIPTION
**Pull request by JasonSinclair95**
changes made to enable sasl for Kafka authentication

### CODE REVIEW

File: charts/corda/templates/_helpers.tpl [hunk 1]
```diff
@@ -141,15 +146,20 @@ Kafka bootstrap servers
...
 {{- define "corda.workerKafkaArgs" -}}
-- -mbootstrap.servers={{ include "corda.kafkaBootstrapServers" . }}
-- --topicPrefix={{ .Values.kafka.topicPrefix }}
+- "-mbootstrap.servers={{ include "corda.kafkaBootstrapServers" . }}"
+- "--topicPrefix={{ .Values.kafka.topicPrefix }}"
 {{- if .Values.kafka.tls.enabled }}
-- -msecurity.protocol=SSL
+{{- if .Values.kafka.sasl.enabled }}
+- "-msecurity.protocol=SASL_SSL"
+- "-msasl.mechanism={{ .Values.kafka.sasl.mechanism }}"
```
Comment (davidcurrie) [inline charts/corda/templates/_helpers.tpl:154]: For completeness, we should also cover the scenario where TLS is not enabled but SASL is (in which case we need to set `security.protocol=SASL_PLAINTEXT` along with the `sasl.mechanism`).

File: charts/corda/templates/sasl-configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{{- if .Values.kafka.sasl.enabled  -}}
+apiVersion: v1
+kind: ConfigMap
```
Comment (davidcurrie) [inline charts/corda/templates/sasl-configmap.yaml]: As per my Slack message, if we're going to embed the password in here, this should be a secret, not a config map.

File: charts/corda/templates/topic-job.yaml [hunk 3]
```diff
@@ -46,7 +46,13 @@ spec:
             - |
                 touch /tmp/working_dir/config.properties
                 {{- if .Values.kafka.tls.enabled }}
+                {{- if .Values.kafka.sasl.enabled }}
```
Comment (davidcurrie) [inline charts/corda/templates/topic-job.yaml:49]: Same again here for support SASL without TLS.

**Review by davidcurrie — CHANGES_REQUESTED** [review]

File: charts/corda/templates/topic-job.yaml [hunk 4]
```diff
@@ -60,6 +60,12 @@ spec:
                 echo "ssl.truststore.password={{ .Values.kafka.tls.truststore.password }}\n" >> /tmp/working_dir/config.properties
                 {{- end }}
                 {{- end }}
+                {{- else }}
+                {{-if .Values.kafka.sasl.enabled }}
+                echo "security.protocol=SASL_SSL\n" >> /tmp/working_dir/config.properties
```
Comment (davidcurrie) [inline charts/corda/templates/topic-job.yaml]:
```suggestion
                echo "security.protocol=SASL_PLAINTEXT\n" >> /tmp/working_dir/config.properties

```

Comment (davidcurrie) [reply charts/corda/templates/topic-job.yaml]: Worth testing this against a prereqs chart with TLS disabled but SASL enabled.

**Review by davidcurrie — CHANGES_REQUESTED** [review]

File: charts/corda/templates/topic-job.yaml (hunk 4 above)
Comment (JasonSinclair95) [reply charts/corda/templates/topic-job.yaml]: This works and all workers start. This was tested by disabling tls in the corda.values then changing the clientprotocal from sasl_tls to sasl in the prereqs.values @davidcurrie

File: charts/corda/templates/flow-worker.yaml [hunk 5]
```diff
@@ -63,7 +63,7 @@ spec:
           httpGet:
             path: /isHealthy
             port: health
-          periodSeconds: 5
+          periodSeconds: 10
```
Comment (davidcurrie) [inline charts/corda/templates/flow-worker.yaml]: I don't think this change should be here - suspect it was a bad merge as it was lowered to 5 in another PR

File: charts/corda/values.yaml [hunk 6]
```diff
@@ -83,6 +83,17 @@ kafka:
...
       password: ""
+  # SASL configuration for client connection to Kafka
+  sasl:
+    # -- SASL username for client connection to Kafka 
+    username: "user"
+    # -- SASL password for client connection to Kafka 
+    password: ""
+    # -- SASL mechanism for client connection to Kafka 
+    mechanism: "SCRAM-SHA-256"
+    # -- enabled/disable SASL for client connection to Kafka
```
Comment (davidcurrie) [inline charts/corda/values.yaml]:
```suggestion
    # -- enable/disable SASL for client connection to Kafka 

```

File: charts/corda/templates/kafka-sasl-secret.yaml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+{{- if .Values.kafka.sasl.enabled  -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "corda.fullname" . }}-sasl
```
Comment (davidcurrie) [inline charts/corda/templates/kafka-sasl-secret.yaml]: I thought we'd renamed this to have `kafka` in the name?

**Review by davidcurrie — CHANGES_REQUESTED** [review]

File: charts/corda/templates/flow-worker.yaml (hunk 5 above)
Comment (JasonSinclair95) [reply charts/corda/templates/flow-worker.yaml]: Okay

File: charts/corda/templates/kafka-sasl-secret.yaml (hunk 7 above)
Comment (JasonSinclair95) [reply charts/corda/templates/kafka-sasl-secret.yaml]: no, but ill add it now

File: charts/corda/templates/_helpers.tpl [hunk 8]
```diff
@@ -225,7 +225,7 @@ Volumes for corda workers
 {{- if .Values.kafka.sasl.enabled  }}
 - name: jaas-conf
   secret:
-    secretName: {{ include "corda.fullname" . }}-sasl
+    secretName: {{ include "corda.fullname" . }}kafka-sasl
```
Comment (davidcurrie) [inline charts/corda/templates/_helpers.tpl]:
```suggestion
    secretName: {{ include "corda.fullname" . }}-kafka-sasl

```

File: charts/corda/templates/flow-worker.yaml (hunk 5 above)
Comment (davidcurrie) [inline charts/corda/templates/flow-worker.yaml]: Still need to set this back to 5...

```suggestion
          periodSeconds: 5
```

File: charts/corda/values.yaml [hunk 9]
```diff
@@ -83,6 +83,18 @@ kafka:
...
+  # SASL configuration for client connection to Kafka
+  sasl:
+    # -- SASL username for client connection to Kafka 
+    username: "user"
+    # -- SASL password for client connection to Kafka 
+    password: ""
+    # -- SASL mechanism for client connection to Kafka 
+    mechanism: "SCRAM-SHA-256"
+    # -- enable/disable SASL for client connection to Kafka 
+
```
Comment (davidcurrie) [inline charts/corda/values.yaml]: Sorry - the extra line was my fault (the GitHub suggestion always adds an extra newline by default for some reason)...

```suggestion
```

**Review by davidcurrie — CHANGES_REQUESTED** [review]

**Review by davidcurrie — APPROVED** [review]
👍

**Pull request merged**